packages/validators/src/schemas/v2023_09_28/experience.ts

```diff
@@ -140,6 +144,7 @@ export const ComponentVariableSchema = z.object({
   defaultValue: PrimitiveValueSchema.or(ComponentPropertyValueSchema).optional(),
   validations: z
     .object({
+      bindingSourceType: BindingSourceTypeEnumSchema,
```

Comment:
Should the `.optional()` be added here instead of in the `BindingSourceTypeEnumSchema` type definition?

That would allow the type to be used in places where we'd want to require a value too.

```suggestion
      bindingSourceType: BindingSourceTypeEnumSchema.optional(),
```

Reply:
Right now since the `BindingSourceTypeEnumSchema` type has `.optional()` it creates a type like this:
```ts
type BindingSourceTypeEnumSchema = 'entry' | 'asset' | 'manual' | 'experience' | undefined;
```

Any time you use the `BindingSourceTypeEnumSchema` type it will always have the possibility of being undefined since it's built into the type.

I suggested removing the `.optional()` from the `BindingSourceTypeEnumSchema` type so that it produces this instead:
```ts
type BindingSourceTypeEnumSchema = 'entry' | 'asset' | 'manual' | 'experience';
```

I feel like that is a better type and more flexible because you can make it optional when you use it:

```ts
bindingSourceType?: BindingSourceTypeEnumSchema
```
or
```ts
bindingSourceType: BindingSourceTypeEnumSchema | undefined
```

Reply:
100% agree.  paired on this and implemented this change.  Even added a bonus of `nonempty()` to the type to disallow consumers providing an empty array.

<img width="885" alt="Screenshot 2025-01-22 at 3 00 58 PM" src="https://github.com/user-attachments/assets/7dff8615-8cc7-4e52-85c4-2e8d34a07ed3" />